Description: Always define TBB_USE_GCC_BUILTINS on GCC.
Author: Adam Conrad <adconrad@ubuntu.com>

--- a/include/tbb/tbb_machine.h
+++ b/include/tbb/tbb_machine.h
@@ -225,6 +225,9 @@
 
 #elif __linux__ || __FreeBSD__ || __NetBSD__ || __OpenBSD__
 
+    #ifndef TBB_USE_GCC_BUILTINS
+        #define TBB_USE_GCC_BUILTINS 1
+    #endif
     #if (TBB_USE_GCC_BUILTINS && __TBB_GCC_BUILTIN_ATOMICS_PRESENT)
         #include "machine/gcc_generic.h"
     #elif (TBB_USE_ICC_BUILTINS && __TBB_ICC_BUILTIN_ATOMICS_PRESENT)
